pydantic/dataclasses.py

```diff
@@ -31,6 +31,7 @@ class M:
 The trick is to create a wrapper around `M` that will act as a proxy to trigger
 validation without altering default `M` behaviour.
 """
+import dataclasses
```

Comment:
due to import time problems in the past, we avoid doing module level import of stuff unless it's absolutely necessary.

I guess if we remove `dataclass` from `__init__.py`, maybe this would be fine, but we'll need to change this or on `v1.10`.

Not sure which you prefer?

Reply:
Found another way.  I moved the overloads and the import into `if TYPE_CHECKING` blocks.

It's actually enough to have one `dataclass_transform` decorator per overload case, even undefined behavior to have multiple. With the unnecessary ones removed, it still seem to works (tested with mypy and pyright).
https://peps.python.org/pep-0681/#undefined-behavior